### PR TITLE
Fix fetchAddOns functionality

### PR DESCRIPTION
**Pull request by goodfellamilos**
There is a not resolved issue in original recurring repo https://github.com/ceejbot/recurring/issues/17
This PR fixes it.

### CODE REVIEW

**Comment by p-janik** [comment]
can we add test for it?

**Comment by goodfellamilos** [comment]
@p-janik Tests added in b880790

File: lib/models/plan.js [hunk 1]
```diff
@@ -61,18 +60,21 @@ class Plan extends RecurlyData {
...
       if (err) {
         return callback(err)
       }
-      this.addons = { }
-      _.each(results, function(addon) {
-        this._addons[addon.add_on_code] = addon
-      })
-      callback(this._addons)
+
+      this._addons = results.reduce((acc, addon) => {
```
Comment (aliismayilov) [inline lib/models/plan.js]: 👍

**Comment by p-janik** [comment]
why not add test like https://github.com/figureaps/recurring/pull/3/files#diff-a5f1f645f200f441a1024dbd8d1ad654 ?

**Comment by goodfellamilos** [comment]
@p-janik Why not :-P It's already there https://github.com/figureaps/recurring/pull/6/files#diff-7c95af2af94b2edf8ca3f62197eb6486

**Pull request merged**